repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -375,16 +383,54 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
     * Converts the saml response document into a servlet input stream for passing down the filter chain.
     *
     * @param document the final saml response
-    * @return the sinput stream that can be wrapped and passed along
+    * @return the input stream that can be wrapped and passed along
     */
-  def convertDocumentToStream(document: Document): ServletInputStream = ???
+  def convertDocumentToStream(document: Document): ServletInputStream = {
+    val source = new DOMSource(document)
+    val baos = new ByteArrayOutputStream()
+    val outputTarget = new StreamResult(baos)
+    transformerFactory.newTransformer().transform(source, outputTarget)
+    new BufferedServletInputStream(new ByteArrayInputStream(baos.toByteArray))
+  }
 
   /**
-    * This is super vague because we don't know what it looks like yet.
+    * Added the extended attributes passed in the SAML Response.
     *
     * @return
     */
-  def doResponseStuff() = ???
+  def addExtendedAttributes(response: HttpServletResponseWrapper, translatedSamlResponse: Document): Option[InputStream] = {
+    logger.debug("Starting the Response processing.")
+    response.getContentType.toLowerCase match {
+      case json if json.contains("json") =>
+        val destination = AttributeMapper.addExtendedAttributes(
+          AttributeMapper.parseJsonNode(new StreamSource(response.getOutputStreamAsInputStream)),
+          translatedSamlResponse,
+          validate = true,
+          XSDEngine.AUTO.toString)
+        val mapper = new ObjectMapper()
+        mapper.getFactory.configure(JsonGenerator.Feature.ESCAPE_NON_ASCII, true)
+        Option(new ByteArrayInputStream(mapper.writeValueAsBytes(destination)))
+      case xml if xml.contains("xml") =>
+        var docBuilder: Option[DocumentBuilder] = None
+        try {
+          docBuilder = Option(XMLParserPool.borrowParser)
+          val outDoc = docBuilder.get.newDocument
+          val destination = new DOMDestination(outDoc)
+          AttributeMapper.addExtendedAttributes(
+            new StreamSource(response.getOutputStreamAsInputStream),
+            new StreamSource(convertDocumentToStream(translatedSamlResponse)),
```

Comment:
you cleaned up the json, but you are doing stuff here still. I suspect we are taking the dom making it a stream, and then it'll get turned back into a dom. I could be wrong though.